spec/model/location_credential_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative "spec_helper"
+
+RSpec.describe LocationCredential do
+  subject(:location_credential) {
+    described_class.create_with_id(location.id, assume_role: "assume-role")
+  }
+
+  let(:location) {
+    Location.create(
+      name: "test-location",
+      display_name: "display-name",
+      ui_name: "ui-name",
+      visible: true,
+      provider: "aws"
+    )
+  }
+
+  it "uses Aws::AssumeRoleCredentials when assume_role set" do
+    creds = instance_double(Aws::AssumeRoleCredentials)
+    expect(Aws::AssumeRoleCredentials).to receive(:new).with(role_arn: "assume-role", role_session_name: Config.aws_role_session_name).and_return(creds)
+    expect(location_credential.credentials).to be(creds)
+  end
+
+  it "gets account id from sts" do
+    skip unless ENV["CLOVER_FREEZE"] != "1"
```

Comment:
`skip` makes the spec show up as pending in the output. It's OK for now, but we should use a conditional around the spec, so the spec (or spec class) is never executed. Could be done in a future PR.